cloud-checksum/src/checksum/aws_etag.rs

```diff
@@ -5,12 +5,83 @@
+/// and SDKs.
+pub const PREFERRED_PART_SIZES: &[u64] = &[
+    // AWS CLI/boto3 uses 8MiB as the default:
+    // https://github.com/aws/aws-cli/blob/b9459db122d9f596a4570b6b5ecca44311b48fc2/awscli/customizations/s3/transferconfig.py#L21
+    // https://github.com/boto/boto3/blob/0442d32d2d2cbc5efbe158f6336993d1ee89b36b/boto3/s3/transfer.py#L243
+    // Java V2 uses 8MiB:
+    // https://github.com/aws/aws-sdk-java-v2/blob/5463dd3d403450e167408895c152c62884da65bd/services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/multipart/MultipartConfigurationResolver.java#L28
+    8 * MIB,
+    // C++ SDK uses 5MiB:
+    // https://github.com/aws/aws-sdk-cpp/blob/93f60cc8aad399a3977287485c19c24d15723b78/src/aws-cpp-sdk-transfer/include/aws/transfer/TransferManager.h#L35
+    // Go seems to also default to 5MiB:
+    // https://github.com/aws/aws-sdk-go-v2/blob/6fa167adb5d1a2618a2f9bbe8f2b885c0d7e2893/feature/s3/transfermanager/options.go#L12-L15
+    // So does JavaScript:
+    // https://github.com/aws/aws-sdk-js-v3/blob/11baa79dff0d9e8fddce873f5306613286990ded/lib/lib-storage/src/Upload.ts#L42
+    // And .NET:
+    // https://github.com/aws/aws-sdk-net/blob/0f72ad1c90f471505c0013c92ec5e9e567239527/sdk/src/Services/S3/Custom/Util/S3Constants.cs#L35
+    // PHP:
+    // https://github.com/aws/aws-sdk-php/blob/713bdb1ff0c2eb519932c111fc450f6b5462b69f/src/S3/MultipartUploader.php#L19
+    5 * MIB,
+    // Console uses 16mib:
+    16 * MIB,
+    // Ruby SDK uses 50MiB:
+    // https://github.com/aws/aws-sdk-ruby/blob/2c8a0686794e8d03da504fcb25984f7fae93f5b3/gems/aws-sdk-s3/lib/aws-sdk-s3/object_multipart_copier.rb#L31
+    50 * MIB,
+    // Java V1 SDK uses 100MiB and 16mib:
+    // https://github.com/aws/aws-sdk-java/blob/bdca0550fc15769618a51338f5f2f84bc603a1cf/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/TransferManagerConfiguration.java#L32-L46
+    // But it also does some calculations based on the total size:
+    // https://github.com/aws/aws-sdk-java/blob/bdca0550fc15769618a51338f5f2f84bc603a1cf/aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/internal/TransferManagerUtils.java#L119-L125
+    100 * MIB,
+    // s3cmd uses 15mib:
+    // https://github.com/s3tools/s3cmd/blob/8cb9b23992714b5ec22c1e514a50996e25aa333b/S3/Config.py#L196
+    15 * MIB,
+    // Some other options that a user might enter:
+    10 * MIB,
+    20 * MIB,
+    200 * MIB,
+    500 * MIB,
+    GIB,
+    2 * GIB,
+    5 * GIB,
+    1000 * MIB,
+    2000 * MIB,
+    5000 * MIB,
+    GB,
+    2 * GB,
+    5 * GB,
+];
+
+/// Find the part size position in `PREFERRED_PART_SIZE`.
+pub const fn part_size_position(part_size: u64) -> Option<usize> {
+    macro_rules! match_index {
+        { $($index:expr),* } => {
+            $(if part_size == PREFERRED_PART_SIZES[$index] { return Some($index + 1) })*
+        }
+    }
+
+    match_index!(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18);
+
+    None
+}
+
```

Comment:
Using a regular array now with proper constants. No need for the compile time map I think, and also the macro doesn't support constants.